sqlite/src/comdb2build.c

```diff
@@ -2244,7 +2246,7 @@ int resolveTableName(struct SrcList_item *p, const char *zDB, char *tableName,
    if ((zDB && (!strcasecmp(zDB, "main") || !strcasecmp(zDB, "temp"))))
    {
        snprintf(tableName, len, "%s", p->zName);
-   } else if (clnt && (clnt->user[0] != '\0') &&
+   } else if (clnt && (clnt->current_user.name[0] != '\0') &&
```

Comment:
Actually, should we use `clnt->current_user.have_name` here to determine whether a username exists? If we assume a username can't be empty, we do not need `have_name`. If a username can be an empty string, `have_name` should be used here.

Reply:
'empty' user name (and password) seems to be working fine except for when user schema is enabled, where I have found a problem. Let me handle that in a separate PR.

Reply:
Yes we use `have_user` and `user[0] != '\0'` interchangeably...